skiko/src/nativeJsMain/cpp/BreakIterator.cc

```diff
@@ -116,11 +120,12 @@ SKIKO_EXPORT KInt* org_jetbrains_skia_BreakIterator__1nGetRuleStatuses
 
 
 SKIKO_EXPORT KNativePointer org_jetbrains_skia_BreakIterator__1nSetText
-  (KNativePointer ptr, KChar* textArr, KInt len) {
+  (KNativePointer ptr, KChar* textArr, KInt len, KInt* errorCode) {
     UBreakIterator* instance = reinterpret_cast<UBreakIterator*>(ptr);
     std::vector<UChar>* text = new std::vector<UChar>(textArr, textArr + len);
 
     UErrorCode status = U_ZERO_ERROR;
     ubrk_setText(instance, text->data(), len, &status);
```

Comment:
Why not just
```
errorCode[0] = U_ZERO_ERROR;
ubrk_setText(instance, text->data(), len, errorCode);
```?

Reply:
@olonho if I got the idea right then, instead of 
```cpp
  UErrorCode status = U_ZERO_ERROR;
    UBreakIterator* instance;
    if (localeStr == nullptr)
      instance = ubrk_open(static_cast<UBreakIteratorType>(type), uloc_getDefault(), nullptr, 0, &status);
    else {
      instance = ubrk_open(static_cast<UBreakIteratorType>(type), skString(localeStr).c_str(), nullptr, 0, &status);
    }

    errorCode[0] = status;
    if (U_FAILURE(status))
      return 0;
    else
      return reinterpret_cast<KNativePointer>(instance);
```

we'll have pretty much something like this
```cpp
errorCode[0] = U_ZERO_ERROR;
    UBreakIterator* instance;
    if (localeStr == nullptr)
      instance = ubrk_open(static_cast<UBreakIteratorType>(type), uloc_getDefault(), nullptr, 0, reinterpret_cast<UErrorCode*>(errorCode));
    else {
      instance = ubrk_open(static_cast<UBreakIteratorType>(type), skString(localeStr).c_str(), nullptr, 0, reinterpret_cast<UErrorCode*>(errorCode));
    }

    if (U_FAILURE(reinterpret_cast<UErrorCode>(errorCode[0])))
      return 0;
    else
      return reinterpret_cast<KNativePointer>(instance);
```

So, for the sake of not having status variable we sacrifice a bit readability with introducing reinterpret_cast and even explicit casting

Reply:
For the JVM, however, it totally make sense - pushed